src/viking.js

```diff
@@ -1,11 +1,86 @@
 // Soldier
-function Soldier() {}
+function Soldier (health, strength) {
+    this.health = health;
+    this.strength = strength;
+    this.attack = function () { 
+        return this.strength;
+    };
+    this.receiveDamage = function (damage) {
+        this.health -= damage;
+    };
+    }
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+
+    Soldier.call(this, health, strength);
+    this.name = name;
+    this.receiveDamage = function (damage) {
+      this.health -= damage;
+      if (this.health > 0) {
+        return `${this.name} has received ${damage} points of damage`;
+      } else {
+        return `${this.name} has died in act of combat`;
+      }
+    };
+    this.battleCry = function () {
+      return `Odin Owns You All!`
+    }
+  }
+  
+// I DON'T UNDERSTAND THE FOLLOWING STATEMENT. It was created by my partner, Henriette, on day 5:
+  Viking.prototype = Object.create(Soldier.prototype);
+  Viking.prototype.constructor = Viking;
```

Comment:
I was going to explain you but you've just asked Uros, if you still don't get it, feel free to come to us and we will explain you